core/src/Kokkos_Macros.hpp

```diff
@@ -124,7 +124,8 @@
 #define KOKKOS_LAMBDA [=]
 #endif
 
-#if (defined(KOKKOS_ENABLE_CXX17) || defined(KOKKOS_ENABLE_CXX20)) && \
+#if (defined(KOKKOS_ENABLE_CXX17) || defined(KOKKOS_ENABLE_CXX20) || \
+     defined(KOKKOS_ENABLE_CXX23)) &&                                \
```

Comment:
Mark it `FIXME_CXX17`

Reply:
All these places are already (supposed to be part of https://github.com/kokkos/kokkos/issues/5295) and I would assume any cleanup after requiring C++17 to search for `KOKKOS_ENABLE_CXX17` anyway.